packages/block-directory/src/store/test/actions.js

```diff
@@ -121,4 +121,48 @@ describe( 'actions', () => {
 			} );
 		} );
 	} );
+
+	describe( 'uninstallBlockType', () => {
+		it( 'should uninstall a block successfully', () => {
+			const generator = uninstallBlockType( item );
+
+			expect( generator.next().value ).toMatchObject( {
+				type: 'API_FETCH',
+			} );
+
+			expect( generator.next( { success: true } ).value ).toEqual( {
+				type: 'REMOVE_INSTALLED_BLOCK_TYPE',
+				item,
+			} );
+
+			expect( generator.next() ).toEqual( {
+				value: undefined,
+				done: true,
+			} );
+		} );
+
+		it( "should set a global notice if the plugin can't uninstall", () => {
+			const generator = uninstallBlockType( item );
+
+			expect( generator.next().value ).toMatchObject( {
+				type: 'API_FETCH',
+			} );
+
+			const apiError = {
+				code: 'could_not_remove_plugin',
+				message: 'Could not fully remove the plugin .',
```

Comment:
I can change this test, but I wanted to avoid changing the API too much since I think (hope) https://github.com/WordPress/gutenberg/pull/22454 will land soon, and we can iterate again on error flows once it is.

Reply:
Yeah if we could minimize or eliminate any changes to the controllers in this PR that'd be awesome :) I'll try my best to make sure that PR lands in the next day or two.

Reply:
Yeah if we could minimize or eliminate any changes to the controllers in this PR that'd be awesome :) I'll try my best to make sure that PR lands in the next day or two.

Reply:
@TimothyBJacobs would it help if I removed the `new WP_Error` fixes from this PR?

Reply:
That would be!

Reply:
👍 I've rebased (to fix the conflict) and dropped the API changes.

Reply:
👍